students/WooseokJ/MailP2fixedWJ.py

```diff
@@ -0,0 +1,111 @@
+
+# coding: utf-8
+
+# In[3]:
+
+
+# print of the main menu
+import sys
+import datetime
+import operator
+
+def menu():
+    print(''''Menu
+    Please choose from below options:
+    1 - Send a Thank You
+    2 - Create a Report
+    3 - Exit the program
+    4 - Print thank you letter to all current doners''')
+#first it will ask to choose a name whoom you would want to send the mail
+#along side that when typing list it will give you all the names on the current list
+#second if the name is not in the list then it will prompt to add the name into the data base
+#lastly it will print the message of the email
+# have a problem here where when asked for the amount of donation if the value type is not a number we get a error how to loop around this?
+def send_a_thankyou():
+    donor_list = []
+    for donor in donor_db:
+        donor_list.append(donor.lower())
+    while True:
+        fullname = input("Enter the full name of the donor typing list will give the names of current doners ")
+        if fullname.lower() == 'list':
+            for donor in donor_db:
+                print(donor)
+        elif fullname.lower() not in donor_list:
+            print(f'{fullname} does not exist in the donor data, '
+                  f'we will add {fullname} to the donor data.')
+            amount = float(input("Please enter the donation amount: "))
+            donor_db.append((fullname, [amount, 1]))
+            print(f"{amount} has been added to {fullname}'s donation history.")
+            break
+        elif fullname.lower() in donor_list:
+            fullname = donor_list.index(fullname.lower())
+            amount = float(input("Please Enter the donation amount: "))
+            donor_db[fullname][1][0] = donor_db[fullname][1][0] + amount
+            donor_db[fullname][1][1] = donor_db[fullname][1][1] + 1
+            print(f"{amount} has been added to {fullname}'s donation history.")
+            break
+    print("Creating a Thank You email:")
+    print(f'Thank you {fullname} for your donation of {amount:^10.2f}!')
+    print()
+    menu()
+#this prints out the database of the donors with their total amaounts and average amount donated
+def create_a_report():
+    sorted_donor_db = sorted(donor_db.items(), key=operator.itemgetter(1), reverse=True)
+    print("Printing report:")
+    title = ('Donor Name', 'Total Given', "Num Gifts", 'Average Gift')
+    print("{:<19} | {:^13} | {:^13} | {:^11}".format(*title))
+    print("-"*70)
+    for i, row in enumerate(donor_db):
```

Comment:
Your code is currently crashing when _create_a_report()_ is called. One thing to change is to simply accessing _donor_db_. _donor_db_ is a dictionary, if you need to access the donors (dictionary keys) and the list of their donations (dictionary values), you just need to do:

for donor, donations in donor_db.items():
    # do something here to print maybe the name of the donor, 
    # the number of donations and the quantity of donations.
    # the name will be in the variable donor, you can do sum(donations)
    # for the total and len(donations) for the quantity.